test/automation/navigation.tests.ps1

```diff
@@ -1209,7 +1209,132 @@ AfterEach{
         test-path .\Bill\BITS | Should be $True
         test-path .\Bill\BITS -Type Container | Should be $False
         test-path .\Bill\BITS -Type Leaf | Should be $True
-    }
+     }
        
  }
+
+Describe "Not Supported Commands test" -Tags "Feature"{
+BeforeEach{
+        cd $home
+        $a=Get-PSDrive -PSProvider SHiPS
+        $a | % {remove-psdrive $_.Name -ErrorAction Ignore}
+
+        $a = new-psdrive -name ss -psprovider SHiPS -root test#Root
+        $a.Name | Should Be "ss"
+
+        cd $testpath
+       }
+
+AfterEach{
+        cd $home
+        $a=Get-PSDrive -PSProvider SHiPS
+        $a | % {remove-psdrive $_.Name -ErrorAction Ignore}
+
+        If(Test-Path $script:homePath) {Remove-Item -Path $script:homePath -force -Recurse -ErrorAction Ignore}
+        
+        cd $testpath
+       }
+ 
+    It "ClearItem throws NotSupported" {
+        cd ss:\William;
+        $null = dir
+        Clear-Item -Path .\Chrisylin -ErrorAction SilentlyContinue -ErrorVariable ev
+        $ev.FullyQualifiedErrorId -match "NotSupported,Microsoft.PowerShell.Commands.ClearItemCommand" | Should Be $true
+       }
+
+    It "SetItem throws NotSupported" {
+        cd ss:\William
+        $null = dir
+        Set-Item -Path .\Chrisylin -Value "what" -ErrorAction SilentlyContinue -ErrorVariable ev
+        $ev.FullyQualifiedErrorId -match "NotSupported,Microsoft.PowerShell.Commands.SetItemCommand" | Should Be $true
+
+       }
+
+    It "MoveItem throws NotSupported" {
+           cd ss:
+           $b = dir ss:\William\Chrisylin
+           $b.GetType().Name -match "ChrisLeaf" | should be $true
+
+           $c = dir ss:\Bill
+           $c.Count -gt 1 | Should be $true
+
+           cd ss:\William
+           $null = dir
+           Move-Item -Path ss:\William\Chrisylin -Destination ss:\Bill -ErrorAction SilentlyContinue -ErrorVariable ev
+           $ev.FullyQualifiedErrorId -match "NotSupported,Microsoft.PowerShell.Commands.MoveItemCommand" | Should Be $true
+        }
+
+    It "CopyItem throws NotSupported" {
+        cd ss:
+        $b = dir ss:\William\Chrisylin
+        $b.GetType().Name -match "ChrisLeaf" | should be $true
+
+        $c = dir ss:\Bill
+        $c.Count -gt 1 | Should be $true
+
+        cd ss:\William
+        $null = dir
+        Copy-Item -Path ss:\William\Chrisylin -Destination ss:\Bill\ -ErrorAction SilentlyContinue -ErrorVariable ev
+        $ev.FullyQualifiedErrorId -match "NotSupported,Microsoft.PowerShell.Commands.CopyItemCommand" | Should Be $true
```

Comment:
This provides a much more helpful message when the test fails. Otherwise it just says "false does not equal true".
```powershell
$ev.FullyQualifiedErrorId | Should BeExactly "NotSupported,Microsoft.PowerShell.Commands.CopyItemCommand"
```

Reply:
Current Format $ev.FullQualifiedErrorId -match "blabla" is easier to manually verify the logic, since "Should BeExactly" is pester only grammar. Plus, you'll have to run the command and get full error details to fix it.
So I'll leave it as is. Thanks.